conan/internal/runner/docker.py

```diff
@@ -0,0 +1,159 @@
+        self.image = profile.runner.get('image')
+        if not (self.dockerfile or self.image):
+            raise ConanException("dockerfile path or docker image name is needed")
+        self.image = self.image or 'conan-runner-default'
+        self.name = f'conan-runner-{profile.runner.get("suffix", "docker")}'
+        self.remove = int(profile.runner.get('remove', 1))
+        self.cache = str(profile.runner.get('cache', 'clean'))
+        self.container = None
+
+    def run(self):
+        """
+        run conan inside a Docker continer
+        """
+        docker_info(f'Building the Docker image: {self.image}')
+        self.build_image()
+        volumes, environment = self.create_runner_environment()
+        docker_info('Creating the docker container')
+        try:
+            if self.docker_client.containers.list(all=True, filters={'name': self.name}):
+                self.container = self.docker_client.containers.get(self.name)
+                self.container.start()
+            else:
+                self.container = self.docker_client.containers.run(
+                    self.image,
+                    "/bin/bash -c 'while true; do sleep 30; done;'",
+                    name=self.name,
+                    volumes=volumes,
+                    environment=environment,
+                    detach=True,
+                    auto_remove=False)
+                self.init_container()
+            self.run_command(self.command)
+            self.update_local_cache()
+        except:
+            ConanOutput().error(f'Something went wrong running "{self.command}" inside the container')
+        finally:
+            if self.container:
+                docker_info('Stopping container')
+                self.container.stop()
+                if self.remove:
+                    docker_info('Removing container')
+                    self.container.remove()
+
+    def build_image(self):
+        if self.dockerfile:
+            docker_build_logs = self.docker_api.build(path=self.dockerfile, tag=self.image)
+            for chunk in docker_build_logs:
+                for line in chunk.decode("utf-8").split('\r\n'):
+                    if line:
+                        stream = json.loads(line).get('stream')
+                        if stream:
+                            ConanOutput().info(stream.strip())
+
+    def run_command(self, command, log=True, stream=True, tty=True):
+        if log:
+            docker_info(f'Running in container: "{command}"')
+        exec_run = self.container.exec_run(f"/bin/bash -c '{command}'", stream=stream, tty=tty)
+        if stream:
+            try:
+                while True:
+                    chunk = next(exec_run.output).decode('utf-8', errors='ignore').strip()
+                    if log:
+                        ConanOutput().info(chunk)
+            except StopIteration:
+                pass
+        else:
+            return exec_run.output.decode('utf-8', errors='ignore').strip()
```

Comment:
Does this already get printed if stream is set to false elsewhere? Or is the idea to have that case be silent?